functions/webhooks/twitter/TwitterToFlex.ts

```diff
@@ -0,0 +1,270 @@
+  TWITTER_ACCESS_TOKEN: string;
+  TWITTER_ACCESS_TOKEN_SECRET: string;
+  CHAT_SERVICE_SID: string;
+  TWITTER_FLEX_FLOW_SID: string;
+};
+
+type MessageCreate = {
+  target: string;
+  recipient_id: string;
+  sender_id: string;
+  message_data: {
+    text: string;
+    entities: {
+      hashtags: string[];
+      symbols: string[];
+      user_mentions: string[];
+      urls: string[];
+    };
+  };
+};
+
+type DirectMessageEvent = {
+  type: string;
+  id: string;
+  created_timestamp: string;
+  message_create: MessageCreate;
+};
+
+export type Body = {
+  direct_message_events?: DirectMessageEvent[];
+  for_user_id?: string;
+  users?: { [key: string]: { name: string; screen_name: string } };
+  xTwitterWebhooksSignature?: string;
+  bodyAsString?: string;
+};
+
+const twitterUniqueNamePrefix = 'twitter:';
+
+/**
+ * Validates that the payload is signed with TWITTER_CONSUMER_SECRET so we know it's comming from Twitter
+ */
+const isValidTwitterPayload = (event: Body, consumerSecret: string): boolean => {
+  if (!event.bodyAsString || !event.xTwitterWebhooksSignature) return false;
+
+  const expectedSignature = crypto
+    .createHmac('sha256', consumerSecret)
+    .update(event.bodyAsString)
+    .digest('base64');
+
+  const isValidRequest = crypto.timingSafeEqual(
+    Buffer.from(event.xTwitterWebhooksSignature),
+    Buffer.from(`sha256=${expectedSignature}`),
+  );
+
+  return isValidRequest;
+};
+
+/**
+ * Retrieves a channel by sid or uniqueName
+ */
+const retrieveChannel = (context: Context<EnvVars>, uniqueName: string) =>
+  context
+    .getTwilioClient()
+    .chat.services(context.CHAT_SERVICE_SID)
+    .channels(uniqueName)
+    .fetch();
+
+/**
+ * Creates a new Flex channel in the Twitter Flex Flow and subscribes webhooks to it's events
+ */
+const createTwitterChannel = async (
+  context: Context<EnvVars>,
+  uniqueChannelName: string,
+  senderId: string,
+  uniqueSenderName: string, // Unique identifier for this person
+  senderScreenName: string, // Twiter handle to show friendly info
+  forUserId: string,
+) => {
+  const twilioNumber = `${twitterUniqueNamePrefix}${forUserId}`;
+
+  const client = context.getTwilioClient();
+
+  const channel = await client.flexApi.channel.create({
+    flexFlowSid: context.TWITTER_FLEX_FLOW_SID,
+    identity: uniqueSenderName,
+    chatUserFriendlyName: senderScreenName,
+    chatFriendlyName: uniqueChannelName,
+    chatUniqueName: uniqueChannelName,
+    target: uniqueChannelName,
+  });
+
+  const channelAttributes = JSON.parse(
+    (
+      await client.chat
+        .services(context.CHAT_SERVICE_SID)
+        .channels(channel.sid)
+        .fetch()
+    ).attributes,
+  );
+
+  await client.chat
+    .services(context.CHAT_SERVICE_SID)
+    .channels(channel.sid)
+    .update({
+      attributes: JSON.stringify({
+        ...channelAttributes,
+        channel_type: 'twitter',
+        twitterUserHandle: senderScreenName,
+        // customChannelType: 'twitter',
```

Comment:
is this commented because we won't need it?

Reply:
Yes, it was a thing I was trying but is not needed in the end. Will remove it when doing cleanup, thanks for the catch!